pkg/odo/cli/component/devfile.go

```diff
@@ -62,7 +62,7 @@ func (po *PushOptions) DevfilePush() (err error) {
 		platformContext = nil
 	} else {
 		kc := kubernetes.KubernetesContext{
-			Namespace: po.namespace,
+			Namespace: po.KClient.Namespace,
```

Comment:
Asking based on this description:
> url create --now, pushes the component and create the url, but it pushes the component in current namespace not in the namespace saved in env.yaml or namespace used while creation.

Wouldn't using `KClient.Namespace` still use the namespace that's currently set? To use the namespace set in `env.yaml`, don't we need to use the env specific info?

Reply:
It already gets resolved while creating context.
https://github.com/openshift/odo/blob/c4aa610fddc3f255e82610d232b13ecc634b91c9/pkg/odo/genericclioptions/context.go#L469

The problem here was if we do direct push `odo push`, it goes through this file
https://github.com/openshift/odo/blob/c4aa610fddc3f255e82610d232b13ecc634b91c9/pkg/odo/cli/component/push.go#L78

and we push via `url create --now`, it skips that and directly calls here
https://github.com/openshift/odo/blob/c4aa610fddc3f255e82610d232b13ecc634b91c9/pkg/odo/cli/component/devfile.go#L36

So po.namespace does not get updated correctly, however we initialize context in `odo url create --now` too, so po.Kclient.namespace always have correct value.

Reply:
OK. So using "env specific info" point was valid. But `KClient.Namespace` already refers that hence this is valid. That's what I understood. :smile: